miniscope_io/data/config/wirefree/sd-layout-battery.yaml

```diff
@@ -0,0 +1,41 @@
+id: wirefree-sd-layout-battery
```

Comment:
Ope responded to this as a global comment before i read the individual comments. Responded to most of these there, but for continuity's sake: currently we get both, and there are reasons to have both, and they don't really trade off. 

> I feel there could be some confusion with duplicate IDs, and ambiguous IDs,

this would be an interface question. so we would want something like

```shell
$ mio config --show
id                    path
sd-layout-battery     miniscope_io/data/configs/sd-layout-battery.yaml
my-config             {user_dir}/configs/my-config.yaml
my-config             {user_dir}/configs/my-config-2.yaml
```

but again you can always specify something as a path if you want to. 

> There will be multiple configs for one device. I think we'll have higher risks of ambiguous configs IDs getting pushed (like ScopeX-1, ScopeX-2, ScopeX-old, ScopeX-latest, and ScopeY for a device called ScopeX), compared to when we just use path/filename as IDs.

the goal is that we don't *want* to have `scope-v1`, `scope-v2`, etc. The idea behind having the `id` is that we want to have a way of identifying unique configurations *semantically*, so then different versions could each be `id: scope-default` with a different `mio_version` field to identify that they are related configs that are for different versions of the scope. Then the id only contains the minimal semantic information necessary to identify it: like one would usually just use `wireless` to run their wireless miniscope, but if they had some custom configuration they could use `wireless-mycustomconfig` everywhere and not have to worry about keeping the rest of their code up to date with the current location of that file, or the different file names as the version changes. 

> Being able to have nice versions in IDs feels nice, but I think it'll be pretty confusing if the filename doesn't sync with it.

Filenames are unique identifiers to the system, but we want to provide an interface that potentially allows people to have multiple versions of the same configuration, and exposing them in the filesystem makes for nicer ux than just having them in a database somewhere. so that's the compromise - have an `id` field and allow it to be decoupled from the filename. 

if someone wants to refer to something by the filename, then they just go `path/filename.yaml`. if someone wants to refer to something by the id, they go `myconfig`. one can make the filename match the if they want to, but they have the ability to make it orthogonal to filename in cases where that's useful.


> There will be multiple configs for one device. I think we'll have higher risks of ambiguous configs IDs getting pushed (like ScopeX-1, ScopeX-2, ScopeX-old, ScopeX-latest, and ScopeY for a device called ScopeX), compared to when we just use path/filename as IDs